services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/geometria/HankeGeometriatDaoImplITest.kt

```diff
@@ -0,0 +1,110 @@
+package fi.hel.haitaton.hanke.geometria
+
+import assertk.assertAll
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNull
+import fi.hel.haitaton.hanke.HaitatonPostgreSQLContainer
+import fi.hel.haitaton.hanke.HankeEntity
+import fi.hel.haitaton.hanke.HankeRepository
+import fi.hel.haitaton.hanke.OBJECT_MAPPER
+import org.geojson.Point
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.jdbc.core.JdbcOperations
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.DynamicPropertyRegistry
+import org.springframework.test.context.DynamicPropertySource
+import org.testcontainers.junit.jupiter.Container
+import org.testcontainers.junit.jupiter.Testcontainers
+import java.nio.file.Files
+import java.nio.file.Paths
+import javax.transaction.Transactional
+
+
+@Testcontainers
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("default")
+@Transactional
+internal class HankeGeometriatDaoImplITest {
+
+    companion object {
+        @Container
+        var container: HaitatonPostgreSQLContainer = HaitatonPostgreSQLContainer
+                .withExposedPorts(5433) // use non-default port
+                .withPassword("test")
+                .withUsername("test")
+
+        @JvmStatic
+        @DynamicPropertySource
+        fun postgresqlProperties(registry: DynamicPropertyRegistry) {
+            registry.add("spring.datasource.url", container::getJdbcUrl)
+            registry.add("spring.datasource.username", container::getUsername)
+            registry.add("spring.datasource.password", container::getPassword)
+            registry.add("spring.liquibase.url", container::getJdbcUrl)
+            registry.add("spring.liquibase.user", container::getUsername)
+            registry.add("spring.liquibase.password", container::getPassword)
+        }
+    }
+
+    @Autowired
+    private lateinit var hankeRepository: HankeRepository
+
+    @Autowired
+    private lateinit var hankeGeometriatDao: HankeGeometriatDao
+
+    @Autowired
+    private lateinit var jdbcOperations: JdbcOperations
+
+    @Test
+    fun `CRUD testing`() {
+        val hankeGeometriat = OBJECT_MAPPER.readValue(Files.readString(Paths.get("src/integrationTest/resources/fi/hel/haitaton/hanke/hankeGeometriat.json")), HankeGeometriat::class.java)
```

Comment:
In order to not mix project directory hierarchy into the code/classpath hiearchy, following can be used:
`val hankeGeometriat = OBJECT_MAPPER.readValue(this::class.java.getResource("/fi/hel/haitaton/hanke/hankeGeometriat.json").readText(Charsets.UTF_8), HankeGeometriat::class.java)`
Alternately, if the json-file is moved to the same package (as it should, ...hanke.geometria/hankeGeometriat.json), it can be made shorter like:
`val hankeGeometriat = OBJECT_MAPPER.readValue(this::class.java.getResource("hankeGeometriat.json").readText(Charsets.UTF_8), HankeGeometriat::class.java)`
(I tested both using the HankeGeometriatServiceImplTest)